.drone.yml

```
@@ -1,32 +1,239 @@
+    when:
+      ref:
+        - refs/heads/master
+        - refs/tags/**
+depends_on:
+  - default
+trigger:
+  ref:
+    - refs/heads/master
+    - refs/tags/**
+    - refs/pull/**
+
+---
+kind: pipeline
+type: vm
+name: manifest
 platform:
   os: linux
   arch: amd64
 pool:
   use: ubuntu
 
 steps:
-- name: build
-  image: golang:1.19
-  commands:
-  - go test ./...
-  - sh scripts/build.sh
-
-- name: publish
-  image: plugins/docker
-  pull: if-not-exists
-  settings:
-    repo: plugins/artifact-metadata-publisher
-    auto_tag: true
-    dockerfile: Dockerfile
-    username:
-      from_secret: docker_username
-    password:
-      from_secret: docker_password
-  when:
-    event:
-      exclude:
-      - pull_request
\ No newline at end of file
+  - name: manifest
+    image: plugins/manifest
+    settings:
+      auto_tag: "true"
+      username:
+        from_secret: docker_username
+      password:
+        from_secret: docker_password
+      spec: docker/manifest.tmpl
+      ignore_missing: true
+depends_on:
+  - linux-amd64
+  - linux-arm64
+  - windows-ltsc2022
+trigger:
+  ref:
+    - refs/heads/master
+    - refs/tags/**
+
+---
+kind: pipeline
+type: vm
+
+pool:
+  use: ubuntu
+
+steps:
+  - name: build
+    pull: always
+    image: golang:1.22
+    commands:
+      - GOOS=linux   GOARCH=amd64   go build -ldflags "-s -w" -a -tags netgo -o release/artifact-metadata-publisher-linux-amd64
+      - GOOS=linux   GOARCH=arm64   go build -ldflags "-s -w" -a -tags netgo -o release/artifact-metadata-publisher-linux-arm64
+      - GOOS=darwin  GOARCH=amd64   go build -ldflags "-s -w" -a -tags netgo -o release/artifact-metadata-publisher-darwin-amd64
+      - GOOS=darwin  GOARCH=arm64   go build -ldflags "-s -w" -a -tags netgo -o release/artifact-metadata-publisher-darwin-arm64
+      - GOOS=windows GOARCH=amd64   go build -ldflags "-s -w" -a -tags netgo -o release/artifact-metadata-publisher-windows-amd64
+
+    environment:
+      CGO_ENABLED: 0
+      GO111MODULE: on
+
+  - name: zstd-compress
+    commands:
+      - sudo apt-get update -y
+      - sudo apt-get install -y zstd
+      - zstd release/artifact-metadata-publisher-linux-amd64
+      - zstd release/artifact-metadata-publisher-linux-arm64
+      - zstd release/artifact-metadata-publisher-darwin-arm64
+      - zstd release/artifact-metadata-publisher-darwin-amd64
+      - zstd release/artifact-metadata-publisher-windows-amd64
+
+  - name: release
+    image: plugins/github-release
+    settings:
+      files:
+        - release/artifact-metadata-publisher-linux-amd64.zst
+        - release/artifact-metadata-publisher-linux-arm64.zst
+        - release/artifact-metadata-publisher-darwin-arm64.zst
+        - release/artifact-metadata-publisher-darwin-amd64.zst
+        - release/artifact-metadata-publisher-windows-amd64.zst
+      api_key:
+        from_secret: github_token
+    when:
+      event:
+        - tag
+depends_on:
+  - linux-amd64
+  - linux-arm64
+  - windows-ltsc2022
```

I think this stage needs the same trigger logic as the `manifest` stage
```suggestion
  - windows-ltsc2022
trigger:
  ref:
    - refs/heads/master
    - refs/tags/**
```